zingolib/src/commands/error.rs

```diff
@@ -2,29 +2,33 @@ use std::fmt;
 
 #[derive(Debug)]
 pub(crate) enum CommandError {
-    FailedJsonParsing(json::Error),
-    FailedIntParsing(std::num::ParseIntError),
-    UnexpectedType,
+    ArgsNotJson(json::Error),
+    SingleArgNotJsonArray(String),
+    ParseIntFromString(std::num::ParseIntError),
+    UnexpectedType(String),
     MissingKey(String),
     InvalidArguments,
     IncompatibleMemo,
     InvalidMemo(String),
+    NonJsonNumberForAmount(String),
 }
 
 impl fmt::Display for CommandError {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         use CommandError::*;
 
         match self {
-            FailedJsonParsing(e) => write!(f, "failed to parse argument. {}", e),
-            FailedIntParsing(e) => write!(f, "failed to parse argument. {}", e),
-            UnexpectedType => write!(f, "arguments cannot be parsed to expected type."),
+            ArgsNotJson(e) => write!(f, "failed to parse argument. {}", e),
+            ParseIntFromString(e) => write!(f, "failed to parse argument. {}", e),
+            UnexpectedType(e) => write!(f, "arguments cannot be parsed to expected type. {}", e),
             MissingKey(key) => write!(f, "json array is missing \"{}\" key.", key),
             InvalidArguments => write!(f, "arguments given are invalid."),
             IncompatibleMemo => {
                 write!(f, "memo's cannot be sent to transparent addresses.")
             }
             InvalidMemo(e) => write!(f, "failed to interpret memo. {}", e),
+            NonJsonNumberForAmount(e) => write!(f, "{}", e),
+            SingleArgNotJsonArray(e) => write!(f, "{}", e),
```

Comment:
Missing display message

```suggestion
            SingleArgNotJsonArray(e) => write!(f, "argument cannot be parsed to a json array: {}", e),
```